tests/test_dimension.py

```diff
@@ -851,6 +851,154 @@ def test_condition_lowering(self):
         Operator(Eq(f[x, ci], g[x, y])).apply()
         assert np.all(f.data == 1)
 
+    def test_strictlessthan_relational(self):
```

Comment:
NOTE for reviewers: The tests below have a lot of code replication, can we reduce throusgh a paremtetrized call?